### PR TITLE
Add LDAP feature CLI support

### PR DESCRIPTION
**Pull request by davidpil2002**
#### What I did
Add LDAP CLI

#### How I did it
created the CLI by using YANG model generator, the YANG model can be found in the LDAP HLD:
https://github.com/sonic-net/SONiC/pull/1487
#### How to verify it
Manually:
you can use configurations command like"config ldap global <policy> <value>"  or 
"show ldap global" (more examples in the HLD.)
Auto:
1.There are unitest of each policy including good & bad flow in this commit, that should pass.
#### Previous command output (if the output of a command-line utility has changed)
No prev commands - New feature
#### New command output (if the output of a command-line utility has changed)
```
sudo show ldap global 
BIND DN                       BIND PASSWORD      BIND TIMEOUT  LDAP VERSION    BASE DN            LDAP PORT      TIMEOUT
----------------------------  ---------------  --------------  --------------  -----------------  -----------  ---------
cn=ldapadm,dc=test1,dc=test2  password            2              3             dc=test1,dc=test2  389                  2

```

### CODE REVIEW

**Comment by liat-grozovik** [comment]
@lguohan who should review and provide feedback?

File: tests/ldap_input/assert_show_output.py [hunk 1]
```diff
@@ -0,0 +1,20 @@
+"""
+Module holding the correct values for show CLI command outputs for the ldap_test.py
+"""
+
+show_ldap_global="""\
+BIND DN                       BIND PASSWORD      BIND TIMEOUT    VERSION  BASE DN              PORT    TIMEOUT
+----------------------------  ---------------  --------------  ---------  -----------------  ------  ---------
+cn=ldapadm,dc=test1,dc=test2  password                      3          3  dc=test1,dc=test2     389          2
```
Comment (madhupalu) [inline tests/ldap_input/assert_show_output.py:8]: what are the units for BIND TIMEOUT and LDAP TIMEOUT?

File: config/plugins/sonic-system-ldap_yang.py [hunk 2]
```diff
@@ -0,0 +1,497 @@
...
+    cfg.setdefault(table, {})
+    if key not in cfg[table]:
+        raise Exception(f"{key} does not exist")
+
+    cfg[table].pop(key)
+
+    validate_config_or_raise(cfg)
+    db.set_entry(table, key, None)
+
+
```
Comment (madhupalu) [inline config/plugins/sonic-system-ldap_yang.py:136]: It is auto generated code, code cleanup must be done.

File: show/plugins/sonic-system-ldap_yang.py [hunk 3]
```diff
@@ -0,0 +1,153 @@
...
+    format_attr_value(
+        entry,
+        {'name': 'priority', 'description': 'Server priority', 'is-leaf-list': False, 'is-mandatory': False, 'group': ''}
+    ),
+]
+
+        body.append(row)
+
+    click.echo(tabulate.tabulate(body, header))
+
```
Comment (madhupalu) [inline show/plugins/sonic-system-ldap_yang.py:64]: Auto generated code, refactor and code clean up must be done

File: tests/ldap_input/assert_show_output.py (hunk 1 above)
Comment (davidpil2002) [reply tests/ldap_input/assert_show_output.py:8]: seconds

File: show/plugins/sonic-system-ldap_yang.py (hunk 3 above)
Comment (davidpil2002) [reply show/plugins/sonic-system-ldap_yang.py:59]: The code was autogenerated by using `sonic-cli-gen` tool
fixed spaces as you mentioned,
pls comment if there is some other issue.
thanks

**Comment by fastiuk** [comment]
@madhupalu , please approve if everything looks good

**Comment by qiluo-msft** [comment]
@madhupalu Do you have more comments?

File: tests/ldap_input/assert_show_output.py (hunk 1 above)
Comment (cbpaviz) [reply tests/ldap_input/assert_show_output.py:8]: Is it possible to mention in the header (secs)?

**Review by liat-grozovik — APPROVED** [review]
In order to merge it @davidpil2002  @Yarden-Z  please check command reference guide and update it with the new additional CLIs. https://github.com/sonic-net/sonic-utilities/blob/master/doc/Command-Reference.md#lldp

**Review by liat-grozovik — CHANGES_REQUESTED** [review]
need to update command reference guide

**Comment by davidpil2002** [comment]
> need to update command reference guide

rebase and update Command-Reference.md file

**Review by liat-grozovik — APPROVED** [review]

**Pull request merged**